### PR TITLE
Fix deleting a data node tier causes load rules to display incorrectly

### PR DESCRIPTION
**Pull request by maytasm**
Fix deleting a data node tier causes load rules to display incorrectly

### Description

When a tier is removed and that tier is currently being used in a rule, the actual rule in the database does not gets modified (rule stay as-is). However, the UI console stops showing the removed tier from the "Edit retention rules" page. Hence, the rules that the user is seeing in the UI will be different than what is actually being run by Druid / stored in the database. This can be very misleading as it hides that the rule is in a bad state (Replicants refers to non-existent tier). We should continue to show the tier even if it does not exist. This will make sure that we are consistent and truthful to the source of truth (a.k.a the metadata store). User can also be aware and change as they realized that the tier that the Replicants is currently set to does not exist.

This PR has:
- [x] been self-reviewed.
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/licenses.yaml)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths.
- [ ] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: web-console/src/components/rule-editor/rule-editor.tsx [hunk 1]
```diff
@@ -102,8 +102,11 @@ export const RuleEditor = React.memo(function RuleEditor(props: RuleEditorProps)
               onChange(RuleUtil.renameTieredReplicants(rule, tier, e.target.value))
             }
           >
+            <option key={tier} value={tier}>
+              {tier}
+            </option>
             {tiers
-              .filter(t => t === tier || !tieredReplicants[t])
+              .filter(t => t !== tier && !tieredReplicants[t])
```
Comment (ccaominh) [inline web-console/src/components/rule-editor/rule-editor.tsx:109]: Can you add a test for this in `rule-editor.spec.tsx`?

Comment (maytasm) [reply web-console/src/components/rule-editor/rule-editor.tsx:109]: Done

**Comment by vogievetsky** [comment]
This looks great, just did some manual testing of this. Thank you for adding it with a test. 
👍 Good to merge after CI

File: web-console/src/components/rule-editor/rule-editor.spec.tsx [hunk 2]
```diff
@@ -36,4 +36,64 @@ describe('rule editor', () => {
...
+          tiers={['test1', 'test2', 'test3']}
+          onChange={() => {}}
+          onDelete={() => {}}
+          moveUp={null}
+          moveDown={null}
+        />
+      );
+      const { container } = render(ruleEditor);
+      expect(container.firstChild).toMatchSnapshot();
+    });
```
Comment (ccaominh) [inline web-console/src/components/rule-editor/rule-editor.spec.tsx:98]: CI is flagging the extra indentation for this test
https://travis-ci.org/github/apache/druid/jobs/689087087#L411

Comment (maytasm) [reply web-console/src/components/rule-editor/rule-editor.spec.tsx]: Done

**Pull request merged**